sunpy/instr/goes.py

```diff
@@ -583,16 +605,490 @@ def _goes_get_chianti_em(longflux, temp, satellite=8, abundances="coronal",
+            raise ValueError("Elements of obstime must be in chronological order.")
+        # Next, get measurement times in seconds from time of first
+        # measurement.
+        obstime_seconds = np.array([(ot-obstime[0]).total_seconds()
+                                    for ot in obstime], dtype="float64")
+        # Finally, integrate using trapezoid rule
+        longlum_int = trapz(longlum.value, obstime_seconds)
+        longlum_int = Quantity(longlum_int, unit=longlum.unit*u.s)
+        shortlum_int = trapz(shortlum.value, obstime_seconds)
+        shortlum_int = Quantity(shortlum_int, unit=shortlum.unit*u.s)
+        # Calculate cumulative radiated energy in each GOES channel as
+        # a function of time.
+        longlum_cumul = cumtrapz(longlum.value, obstime_seconds)
+        longlum_cumul = Quantity(longlum_cumul, unit=longlum.unit/u.s)
+        shortlum_cumul = cumtrapz(shortlum.value, obstime_seconds)
+        shortlum_cumul = Quantity(shortlum_cumul,
+                                  unit=shortlum.unit/u.s)
+        lx_out = {"longlum":longlum, "shortlum":shortlum,
+                  "longlum_cumul":longlum_cumul,
+                  "shortlum_cumul":shortlum_cumul,
+                  "longlum_int":longlum_int, "shortlum_int":shortlum_int}
+    else:
+        lx_out = {"longlum":longlum, "shortlum":shortlum}
+
+    return lx_out
+
+@quantity_input(flux=u.W/u.m/u.m)
+def _calc_xraylum(flux, date=None):
+    """
+    Calculates solar luminosity based on observed flux observed at 1AU.
+
+    This function calculates the luminosity from the Sun based
+    on observed flux in W/m**2.  The units of the results are erg/s.
+    The calculation is made by simply assuming that the radiation is
+    emitted isotropically, i.e. is distributed over a spherical
+    surface area with a radius equal to the Sun-Earth distance.
+
+    Parameters
+    ----------
+    flux : `astropy.units.Quantity`
+       Containing the observed solar flux.  Units=[W/m**2]
+
+    date : (optional) datetime object or valid date string
+        Used to calculate a more accurate Sun-Earth distance based on
+        Earth's orbit at that date.  If date is not set, standard value
+        for 1AU used.
+
+    Returns
+    -------
+    xraylum : numpy array, dtype=float, units=erg/s.
+        Array of X-ray luminosity.
+
+    Notes
+    -----
+    To convert from W/m**2 to erg/s:
+    1 W = 1 J/s = 10**7 erg/s
+    1 W/m**2 = 4*pi * AU**2 * 10**7 erg/s, where AU is the Sun-Earth
+    distance in metres.
+
+    Examples
+    --------
+    >>> flux = np.array([7e-6,7e-6])
+    >>> xraylum = _calc_xraylum(flux, date="2014-04-21")
+    >>> xraylum
+    array([  1.98649103e+25,   1.98649103e+25])
+
+    """
+    if date is not None:
+        date = parse_time(date)
+        xraylum = 4 * np.pi * (sun.sunearth_distance(t=date)*sun.constants.au)**2 * flux
+    else:
+        xraylum = 4 * np.pi * (sun.sunearth_distance()*sun.constants.au)**2 * flux
```

Comment:
would this just not work without the if?, so if we pass `t=date` that it's `None`, would not work?